src/force-move.ts

```diff
@@ -0,0 +1,229 @@
+  const appPartHash = hashAppPart(lastState);
+  const numStates = states.length;
+
+  const {outcome} = challengeState;
+  const challengeOutcomeHash = hashOutcome(outcome);
+
+  const challengerAddress = participants[challengeState.turnNum % participants.length];
+  const channelStorageLiteBytes = encodeChannelStorageLite({
+    outcome,
+    finalizesAt,
+    state: challengeState,
+    challengerAddress,
+  });
+
+  const data = ForceMoveContractInterface.functions.concludeFromChallenge.encode([
+    turnNumRecord,
+    largestTurnNum,
+    fixedPart,
+    appPartHash,
+    numStates,
+    whoSignedWhat,
+    signatures,
+    challengeOutcomeHash,
+    channelStorageLiteBytes,
+  ]);
+  return {data, gasLimit: GAS_LIMIT};
+}
+
+export function createConcludeFromOpenTransaction(
+  turnNumRecord: number,
+  states: State[],
+  signatures: Signature[],
+  whoSignedWhat: number[],
+): TransactionRequest {
+  // Sanity checks on expected lengths
+  if (states.length === 0) {
+    throw new Error('No states provided');
+  }
+  const {participants} = states[0].channel;
+  if (participants.length !== signatures.length) {
+    throw new Error(
+      `Participants (length:${participants.length}) and signatures (length:${signatures.length}) need to be the same length`,
+    );
+  }
+
+  const lastState = states.reduce((s1, s2) => (s1.turnNum >= s2.turnNum ? s1 : s2), states[0]);
+  const largestTurnNum = lastState.turnNum;
+  const fixedPart = getFixedPart(lastState);
+  const appPartHash = hashAppPart(lastState);
+
+  const outcomeHash = hashOutcome(lastState.outcome);
+
+  const numStates = states.length;
+
+  const data = ForceMoveContractInterface.functions.concludeFromOpen.encode([
+    turnNumRecord,
+    largestTurnNum,
+    fixedPart,
+    appPartHash,
+    outcomeHash,
+    numStates,
+    whoSignedWhat,
+    signatures,
+  ]);
+  return {data, gasLimit: GAS_LIMIT};
+}
+
+export function createRefuteTransaction(
+  turnNumRecord: number,
+  finalizesAt: string,
+  challengeState: State,
+  refuteState: State,
+  refutationStateSignature: Signature,
+): TransactionRequest {
+  const {participants} = challengeState.channel;
+  const variablePartAB = [getVariablePart(challengeState), getVariablePart(refuteState)];
+  const fixedPart = getFixedPart(refuteState);
+  const isFinalAB = [challengeState.isFinal, refuteState.isFinal];
+
+  const challengerAddress = participants[challengeState.turnNum % participants.length];
+  const refutationStateTurnNum = refuteState.turnNum;
+
+  const data = ForceMoveContractInterface.functions.refute.encode([
+    turnNumRecord,
+    refutationStateTurnNum,
+    finalizesAt,
+    challengerAddress,
+    isFinalAB,
+    fixedPart,
+    variablePartAB,
+    refutationStateSignature,
+  ]);
+  return {data, gasLimit: GAS_LIMIT};
+}
+
+export function createForceMoveTransaction(
+  turnNumRecord: number,
+  states: State[],
+  signatures: Signature[],
+  whoSignedWhat: number[],
+  challengerSignature: Signature,
```

Comment:
I'd probably want this method to be passed a signing key and create the signature.